dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/merge/MergeProcessor.java

```diff
@@ -0,0 +1,773 @@
+    }
+
+    private static void removeContextRoot(Server server, String contextRoot) {
+        String url = server.getUrl();
+        String newUrl;
+        if (url.endsWith("/")) {
+            newUrl = url.substring(0, url.length() - contextRoot.length() - 1);
+        } else {
+            newUrl = url.substring(0, url.length() - contextRoot.length());
+        }
+        server.setUrl(newUrl);
+    }
+
+    private static boolean serverEndsWithContextRoot(Server server, String contextRoot) {
+        return server.getUrl().endsWith(contextRoot) || server.getUrl().endsWith(contextRoot + "/");
+    }
+
+    private static boolean isSecurityIdentical(List<InProgressModel> models) {
+        List<List<SecurityRequirement>> reqs = models.stream().map(d -> d.model.getSecurity()).collect(toList());
+
+        return allEqual(reqs, ModelEquality::equals);
+    }
+
+    private static boolean isServersIdentical(List<InProgressModel> processedModels) {
+        List<List<Server>> servers = processedModels.stream().map(m -> m.model).map(OpenAPI::getServers).collect(toList());
+        return allEqual(servers, ModelEquality::equals);
+    }
+
+    private static boolean isInfoIdentical(List<InProgressModel> models) {
+        List<Info> infos = models.stream().map(m -> m.model).map(OpenAPI::getInfo).collect(toList());
+        return allEqual(infos, ModelEquality::equals);
+    }
+
+    private static boolean isExternalDocsIdentical(List<InProgressModel> models) {
+        List<ExternalDocumentation> docs = models.stream().map(m -> m.model).map(OpenAPI::getExternalDocs).collect(toList());
+        return allEqual(docs, ModelEquality::equals);
+    }
+
+    private static void moveServersUnderPaths(OpenAPI model) {
+        List<Server> servers = model.getServers();
+        if (servers == null) {
+            return;
+        }
+
+        Paths paths = model.getPaths();
+
+        if (paths != null) {
+            for (PathItem item : notNull(paths.getPathItems()).values()) {
+                if (item.getServers() == null) {
+                    item.setServers(servers);
+                }
+            }
+        }
+
+        model.setServers(null);
+    }
+
+    /**
+     * Check whether all elements of {@code collection} are equal to each other using the given equality function
+     * <p>
+     * Actually assumes that equals is implemented properly and just checks that the first element is equal to all others
+     * <p>
+     * If {@code collection} contains less than two elements, this method will always return {@code true}.
+     * 
+     * @param <T> the element type
+     * @param collection the collection of elements to test for equality
+     * @param comparator the function to use to test equality
+     * @return {@code true} if all elements of {@code collection} are equal, {@code false} otherwise
+     */
+    private static <T> boolean allEqual(Collection<? extends T> collection, BiPredicate<? super T, ? super T> comparator) {
```

Comment:
should these 4 static methods be in a generic utility class?

Reply:
I wasn't sure. They are very generic, but on the other hand I'm not planning to use them anywhere else right now. I would definitely move them to a utility class if I had somewhere else I wanted to use them.

Reply:
Moved these methods to `OpenAPIUtils`